common/flags/tests/FlagsTest.java

```diff
@@ -232,7 +232,199 @@ public void testRequiredStringFlagWithValue() throws Exception {
   }
 
   @Test(expected = IllegalArgumentException.class)
-  public void testFlagThrowsWithoutParsing() throws Exception {
+  public void testFlagThrowsWithoutParsing() {
     FlagDescTestClass.integerFlag.get();
   }
+
+  @Test
```

Comment:
No need to duplicate all tests, since most of the functionality is already tested.
Since you added code that takes flags from a class, here are some things I'd test:
* Find flag in class
* Not fail when there is no flag in class
* Not take flag in package but in another class